usr/src/cmd/zfs/zfs_main.c

```diff
@@ -3769,15 +3769,42 @@ zfs_do_send(int argc, char **argv)
 			flags.compress = B_TRUE;
 			break;
 		case ':':
-			(void) fprintf(stderr, gettext("missing argument for "
-			    "'%c' option\n"), optopt);
+			/*
+			 * If a parameter was not passed, optopt contains the
+			 * value that would normally lead us into the
+			 * appropriate case statement.  If it's > 256, then this
+			 * must be a longopt and we should look at argv to get
+			 * the string.  Otherwise it's just the character, so we
+			 * should use it directly.
+			 */
+			if (optopt <= UINT8_MAX) {
```

Comment:
All of the options today are, but the customary way to create options that have long flags and no short flags is to use a value greater than 256.  I have a couple patches that I plan to upstream soonish that add longopts with no short version.  This statement is intended to handle that case; any flag that has a short version will use it here (because the value in column 4 will be an ascii character, which is < 255), and any flags that don't will be looked up using the optind.